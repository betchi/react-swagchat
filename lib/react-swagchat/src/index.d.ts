/// <reference types="react-router-dom" />
export { PluginMessageText, PluginMessageImage } from './plugins/message';
export interface IContext {
    router: Router;
    client: Client;
    uiSettings: IUISettings;
}
export interface IOnClickProps {
    onClick?: (param?: any) => void;
}
export interface IIconProps {
    className?: string;
    style?: Object;
}
export declare const logColor = "#3F51B5";
import { HashRouter as Router } from 'react-router-dom';
import { Client, IUISettings } from 'swagchat-sdk';
