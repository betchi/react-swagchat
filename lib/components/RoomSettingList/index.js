"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var RoomSettingList_1 = require("./RoomSettingList");
exports.RoomSettingList = RoomSettingList_1.RoomSettingList;
var RoomSettingModalItem_1 = require("./RoomSettingModalItem");
exports.RoomSettingModalItem = RoomSettingModalItem_1.RoomSettingModalItem;
//# sourceMappingURL=index.js.map