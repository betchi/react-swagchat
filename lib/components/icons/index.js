"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AddCircle_1 = require("./AddCircle");
exports.AddCircle = AddCircle_1.AddCircle;
var Back_1 = require("./Back");
exports.Back = Back_1.Back;
var Block_1 = require("./Block");
exports.Block = Block_1.Block;
var Close_1 = require("./Close");
exports.Close = Close_1.Close;
var Exit_1 = require("./Exit");
exports.Exit = Exit_1.Exit;
var Expand_1 = require("./Expand");
exports.Expand = Expand_1.Expand;
var CheckCircle_1 = require("./CheckCircle");
exports.CheckCircle = CheckCircle_1.CheckCircle;
var Keyboard_1 = require("./Keyboard");
exports.Keyboard = Keyboard_1.Keyboard;
var Camera_1 = require("./Camera");
exports.Camera = Camera_1.Camera;
var Send_1 = require("./Send");
exports.Send = Send_1.Send;
//# sourceMappingURL=index.js.map