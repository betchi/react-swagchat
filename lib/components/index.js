"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Avatar_1 = require("./Avatar/Avatar");
exports.Avatar = Avatar_1.Avatar;
var Badge_1 = require("./Badge/Badge");
exports.Badge = Badge_1.Badge;
var Button_1 = require("./Button/Button");
exports.Button = Button_1.Button;
var CheckListItem_1 = require("./CheckListItem/CheckListItem");
exports.CheckListItem = CheckListItem_1.CheckListItem;
var ContactList_1 = require("./ContactList/ContactList");
exports.ContactList = ContactList_1.ContactList;
var AddCircle_1 = require("./icons/AddCircle");
exports.AddCircle = AddCircle_1.AddCircle;
var Back_1 = require("./icons/Back");
exports.Back = Back_1.Back;
var Block_1 = require("./icons/Block");
exports.Block = Block_1.Block;
var Camera_1 = require("./icons/Camera");
exports.Camera = Camera_1.Camera;
var CheckCircle_1 = require("./icons/CheckCircle");
exports.CheckCircle = CheckCircle_1.CheckCircle;
var CheckCircleOutline_1 = require("./icons/CheckCircleOutline");
exports.CheckCircleOutline = CheckCircleOutline_1.CheckCircleOutline;
var Close_1 = require("./icons/Close");
exports.Close = Close_1.Close;
var Done_1 = require("./icons/Done");
exports.Done = Done_1.Done;
var Edit_1 = require("./icons/Edit");
exports.Edit = Edit_1.Edit;
var Exit_1 = require("./icons/Exit");
exports.Exit = Exit_1.Exit;
var Expand_1 = require("./icons/Expand");
exports.Expand = Expand_1.Expand;
var Keyboard_1 = require("./icons/Keyboard");
exports.Keyboard = Keyboard_1.Keyboard;
var Photo_1 = require("./icons/Photo");
exports.Photo = Photo_1.Photo;
var RadioButtonChecked_1 = require("./icons/RadioButtonChecked");
exports.RadioButtonChecked = RadioButtonChecked_1.RadioButtonChecked;
var RadioButtonUnChecked_1 = require("./icons/RadioButtonUnChecked");
exports.RadioButtonUnChecked = RadioButtonUnChecked_1.RadioButtonUnChecked;
var Send_1 = require("./icons/Send");
exports.Send = Send_1.Send;
var MessageBody_1 = require("./Message/MessageBody");
exports.MessageBody = MessageBody_1.MessageBody;
var MessageDateSeparator_1 = require("./Message/MessageDateSeparator");
exports.MessageDateSeparator = MessageDateSeparator_1.MessageDateSeparator;
var MessageInteractionBottom_1 = require("./Message/MessageInteractionBottom");
exports.MessageInteractionBottom = MessageInteractionBottom_1.MessageInteractionBottom;
var MessageInteractionTop_1 = require("./Message/MessageInteractionTop");
exports.MessageInteractionTop = MessageInteractionTop_1.MessageInteractionTop;
var MessageItem_1 = require("./Message/MessageItem");
exports.MessageItem = MessageItem_1.MessageItem;
var MessageMenuBottom_1 = require("./Message/MessageMenuBottom");
exports.MessageMenuBottom = MessageMenuBottom_1.MessageMenuBottom;
var MessageMenuTop_1 = require("./Message/MessageMenuTop");
exports.MessageMenuTop = MessageMenuTop_1.MessageMenuTop;
var ModalDialog_1 = require("./Modal/ModalDialog");
exports.ModalDialog = ModalDialog_1.ModalDialog;
var ModalView_1 = require("./Modal/ModalView");
exports.ModalView = ModalView_1.ModalView;
var PhotoEdit_1 = require("./PhotoEdit/PhotoEdit");
exports.PhotoEdit = PhotoEdit_1.PhotoEdit;
var RoomEdit_1 = require("./RoomEdit/RoomEdit");
exports.RoomEdit = RoomEdit_1.RoomEdit;
var RoomItem_1 = require("./RoomList/RoomItem");
exports.RoomItem = RoomItem_1.RoomItem;
var RoomList_1 = require("./RoomList/RoomList");
exports.RoomList = RoomList_1.RoomList;
var RoomSettingList_1 = require("./RoomSettingList/RoomSettingList");
exports.RoomSettingList = RoomSettingList_1.RoomSettingList;
var IconListItem_1 = require("./SimpleListItem/IconListItem");
exports.IconListItem = IconListItem_1.IconListItem;
var SimpleListItem_1 = require("./SimpleListItem/SimpleListItem");
exports.SimpleListItem = SimpleListItem_1.SimpleListItem;
var SubTitleBar_1 = require("./SubTitleBar/SubTitleBar");
exports.SubTitleBar = SubTitleBar_1.SubTitleBar;
var TopBar_1 = require("./TopBar/TopBar");
exports.TopBar = TopBar_1.TopBar;
//# sourceMappingURL=index.js.map