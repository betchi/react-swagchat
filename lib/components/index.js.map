{"version":3,"file":"index.js","sourceRoot":"","sources":["../../src/components/index.ts"],"names":[],"mappings":";;AAAA,0CAAyC;AAAhC,0BAAA,MAAM,CAAA;AACf,uCAAsC;AAA7B,wBAAA,KAAK,CAAA;AACd,+CAA8C;AAArC,gCAAA,SAAS,CAAA;AAClB,mDAAgE;AAAvD,oCAAA,WAAW,CAAA;AACpB,0CAAyC;AAAhC,0BAAA,MAAM,CAAA;AACf,yDAAwD;AAA/C,oCAAA,WAAW,CAAA;AACpB,kEAAiE;AAAxD,0CAAA,cAAc,CAAA;AACvB,8DAA6D;AAApD,sCAAA,YAAY,CAAA;AACrB,gDAA+D;AAAtC,8BAAA,QAAQ,CAAA;AACjC,gDAA+C;AAAtC,8BAAA,QAAQ,CAAA;AACjB,0CAAyC;AAAhC,0BAAA,MAAM,CAAA;AACf,yDAA2E;AAA/C,oCAAA,WAAW,CAAA;AACvC,+DAA8D;AAArD,wCAAA,aAAa,CAAA;AACtB,uEAAsE;AAA7D,sDAAA,oBAAoB,CAAA;AAC7B,qDAAoD;AAA3C,oCAAA,WAAW,CAAA;AACpB,qDAAoD;AAA3C,oCAAA,WAAW,CAAA;AACpB,mEAAkE;AAAzD,kDAAA,kBAAkB,CAAA;AAC3B,qDAAoD;AAA3C,oCAAA,WAAW,CAAA;AACpB,qEAA2F;AAA3D,4CAAA,eAAe,CAAA;AAC/C,gDAA+C;AAAtC,8BAAA,QAAQ,CAAA;AACjB,mDAAkD;AAAzC,gCAAA,SAAS,CAAA"}