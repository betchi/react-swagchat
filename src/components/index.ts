export { Avatar } from './Avatar/Avatar';
export { Badge } from './Badge/Badge';
export { Button } from './Button/Button';
export { ContactList, IContactListProps } from './ContactList/ContactList';
export { MessageBody } from './Message/MessageBody';
export { MessageInteraction } from './Message/internal/MessageInteraction';
export { MessageItem } from './Message/internal/MessageItem';
export { MessageMenu } from './Message/internal/MessageMenu';
export { Modal } from './Modal/Modal';
export { ModalAction, IModalAction } from './Modal/ModalAction';
export { PhotoEdit } from './PhotoEdit/PhotoEdit';
export { RoomList, IRoomListProps } from './RoomList/RoomList';
export { RoomSettingButtons, IRoomSettingButtonsProps } from './RoomSettingButtons/RoomSettingButtons';
export { SubTitleBar } from './SubTitleBar/SubTitleBar';
export { TextAvatar } from './TextAvatar/TextAvatar';
export { TopBar } from './TopBar/TopBar';

export interface IOnClickProps {
  onClick?: (param?: any) => void;
}

export interface IRootStyleProps {
  className?: string;
  style?: Object;
}