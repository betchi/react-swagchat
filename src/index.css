@import './pages/**/*.css';
@import './components/**/*.css';
@import './containers/**/*.css';
@import './plugins/**/**/*.css';

body {
  font-family: 'Slabo 13px', 'Rounded Mplus 1c', serif;
}
.nodata-wrap {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 500px;
}
.nodata-text {
  opacity: 0.4;
}
.nodata-image {
  width: 80%;
  opacity: 0.3;
  position: fixed;
}
:placeholder-shown,
::-webkit-input-placeholder {
  color: #cccccc;
}

.layout-box-1 {
  padding: 10px 0;
  background-color: #fafafa;
}

.page-container {
  min-width: 320px;
}

.button {
  border: 10px;
  box-sizing: border-box;
  display: inline-block;
  font-family: Roboto, sans-serif;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  cursor: pointer;
  text-decoration: none;
  padding: 5px;
  outline: none;
  font-size: 0px;
  font-weight: inherit;
  position: relative;
  z-index: 1;
  overflow: visible;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  width: 38px;
  height: 38px;
  background: none;
}
.icon {
  display: inline-block;
  color: #0084ff;
  fill: currentcolor;
  user-select: none;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
}
.button_text {
  color: #0084ff;
  font-size: 24px;
  width: 50px;
  height: 38px;
}