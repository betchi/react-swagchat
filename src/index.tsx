import {
  renderTemplateMessenger,
  renderTemplateSimpleMessage,
} from './templates/';
import {
  Avatar,
  Badge,
  Button,
  ContactList,
  MessageBody,
  Modal,
  ModalAction,
  PhotoEdit,
  RoomEditForm,
  RoomList,
  RoomSettingButtons,
  SubTitleBar,
  TextAvatar,
  TopBar,
} from './components';
import {
  PluginMessageText,
  PluginMessageImage,
  TextItem,
  ImageItem,
} from './plugins/message';
import {
  PluginRoomListItemRoomAndUserNameWithMessage,
  PluginRoomListItemRoomNameWithMessage,
} from './plugins/roomListItem';

export * from './components';
export * from './templates';

// For browser
(window as any).Swag = {};

(window as any).Swag.renderTemplateMessenger = renderTemplateMessenger;
(window as any).Swag.renderTemplateSimpleMessage = renderTemplateSimpleMessage;

(window as any).Swag.Avatar = Avatar;
(window as any).Swag.Badge = Badge;
(window as any).Swag.Button = Button;
(window as any).Swag.ContactList = ContactList;
(window as any).Swag.MessageBody = MessageBody;
(window as any).Swag.Modal = Modal;
(window as any).Swag.ModalAction = ModalAction;
(window as any).Swag.PhotoEdit = PhotoEdit;
(window as any).Swag.RoomEditForm = RoomEditForm;
(window as any).Swag.RoomList = RoomList;
(window as any).Swag.RoomSettingButtons = RoomSettingButtons;
(window as any).Swag.SubTitleBar = SubTitleBar;
(window as any).Swag.TextAvatar = TextAvatar;
(window as any).Swag.TopBar = TopBar;

(window as any).Swag.PluginRoomListItemRoomAndUserNameWithMessage = PluginRoomListItemRoomAndUserNameWithMessage;
(window as any).Swag.PluginRoomListItemRoomNameWithMessage = PluginRoomListItemRoomNameWithMessage;
(window as any).Swag.PluginMessageText = PluginMessageText;
(window as any).Swag.PluginMessageImage = PluginMessageImage;
(window as any).Swag.TextItem = TextItem;
(window as any).Swag.ImageItem = ImageItem;
