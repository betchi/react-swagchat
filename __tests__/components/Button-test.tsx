declare var jest, describe, it, expect, require;

import * as React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import { Button } from '../../src/components';

const testComponent1 = shallow(
  <Button
    text="Button"
  />
);
it('Button renders correctly. Very simple setting.', () => {
  const j = toJson(testComponent1);
  expect(j).toMatchSnapshot();
});

const testComponent2 = shallow(
  <Button
    text="Button"
    icon={<div />}
    shape="round"
    color="secondary"
    position="left"
    iconPosition="right"
    className="test-class"
    style={{width: '100px'}}
    onClick={() => console.log('click')}
  />
);
it('Button renders correctly. All property setting.', () => {
  const j = toJson(testComponent2);
  expect(j).toMatchSnapshot();
});
