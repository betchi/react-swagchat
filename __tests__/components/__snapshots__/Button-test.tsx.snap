// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly. All property setting. 1`] = `
<a
  className="test-class"
  onClick={[Function]}
>
  <div
    style={
      Object {
        "justifyContent": "flex-start",
      }
    }
  >
    <div
      className="text"
    >
      Button
    </div>
    <div
      className="icon"
    >
      <div />
    </div>
  </div>
</a>
`;

exports[`Button renders correctly. Very simple setting. 1`] = `
<a
  className=""
  onClick={[Function]}
>
  <div
    style={
      Object {
        "justifyContent": "flex-start",
      }
    }
  >
    <div
      className="text"
    >
      Button
    </div>
  </div>
</a>
`;
